### ISSUE TITLE
merkaartor 0.18 seems to allow empty tag value?

### ISSUE THREAD

**Issue by GerdP** [openstreetmap/merkaartor#86]
Hi!
I found this changeset
https://www.openstreetmap.org/changeset/36612297
where this node
https://www.openstreetmap.org/node/3718637401
was changed with 
highway=""
I think there is no good reason to safe an empty tag value.

ciao,
Gerd

**Comment by Krakonos** [comment]
Yes. I think this was a decision when I was redoing some logic in the tag manager.

In the 0.19 release, I plan to implement basic rules check before upload. This can be a thing to check, which would be a solution. Other solutions are:

1) Delete the key as well as the tag if it's empty (I think this was the behavior before I changed it)
2) Show dialog expaining what happened, possibly not allowing to finish the editation until it's either canceled or filled in (I would like to avoid necessary dialogs).
3) Allow it, but show the tag name in red, perhaps with a tooltip informing it's not allowed.

I'd prefer solution 3, as it will fit nicely into the rules check I will implement later. I also would like to avoid adding features now, because I'm working on a 0.18.3 release (currently, I think all is done, I'm just trying to get Mac OS X packaging to work).

What do you think?

**Comment by GerdP** [comment]
I like the way JOSM does it:
When you select a tag for editing, the dialog box shows the hint "(An empty value deletes the tag)"
and that is what should happen IMHO

**Comment by vlivyur** [comment]
[Here](https://www.openstreetmap.org/node/11165618593) too in v1. And [here](https://www.openstreetmap.org/way/9222414) in v18
